### PR TITLE
Feature/add organization

### PR DESCRIPTION
**Pull request by GunnarAtEgde**
Added code to get organisation data from NeTEx compatible organisation API endpoint and use that data to provide organisation list via GraphQL endpoint

### Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (changes to documentation only)
- [ ] Other (please describe)

### Issue

#110

### CODE REVIEW

File: sobek-common/src/main/java/org/rutebanken/sobek/organisation/NetexPublicationDeliveryOrganisationRegistry.java [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+    );
+    private final NetexUnmarshaller netexUnmarshaller = new NetexUnmarshaller(
+            PublicationDeliveryStructure.class
+    );
+
+    private final List<Organisation_VersionStructure> organisations = Collections.synchronizedList(
+            new ArrayList<>()
+    );
+    
+    private volatile Instant lastLoadTime;
```
[the remark this answers is not in the document]
Comment (GunnarAtEgde) [reply sobek-common/src/main/java/org/rutebanken/sobek/organisation/NetexPublicationDeliveryOrganisationRegistry.java]: @copilot apply changes based on [this feedback](https://github.com/entur/sobek/pull/115#discussion_r3188230254)

**Comment by dynnamitt** [comment]
# Claude Code review

## Minimum to merge

1. **Fix `getOperator` (type bug, will throw in production)** — `sobek-common/.../NetexPublicationDeliveryOrganisationRegistry.java` `getOperator(String)` filters by `id` only and casts to `Operator`. The fixture has Authority + Operator pairs sharing `organisationId`; if the Authority is hit first the cast `ClassCastException`s. Add `&& org instanceof Operator` like `getAuthority` does.

2. **Fix `GeneralOrganisation → OTHER` mapping** — `sobek-repository/.../OrganisationRepositoryImpl.java` `getOrganisationType(...)`. Registry already exposes `getGeneralOrganisations()`; map this to a `GENERAL_ORGANISATION` enum value instead of `OTHER`.

3. **Catch broadly in `loadOrganisations` and update `lastLoadTime` on failure** — `NetexPublicationDeliveryOrganisationRegistry.java` `loadOrganisations()` only catches `NetexUnmarshallerUnmarshalFromSourceException`. The `forEach` throws `UnsupportedOrganisationTypeException` (RuntimeException) which propagates out, and combined with `lastLoadTime` not being set on failure, every subsequent getter retries and re-throws. Catch broadly + always update `lastLoadTime`.

4. **Revert or split out the H2 dialect change** — `sobek-service/src/test/resources/application.properties:2,49`: switching to `MODE=MySQL` and commenting out `PostgreSQLDialect` lets Postgres-only bugs slip through. Production is PostGIS. Revert or split.

5. **Decouple HTTP fetch from the read-path lock** (or at minimum, keep stale data during refresh) — `NetexPublicationDeliveryOrganisationRegistry.java` `ensureFreshData()` synchronizes around `loadOrganisations()`, which `block(Duration.ofSeconds(30))`s in `NetexPublicationDeliveryHttpOrganisationRegistry.java` `getPublicationDeliverySource()`. A slow upstream stalls every concurrent reader for up to 30s. Either keep the previous cache visible during refresh, or move refresh off the read path.

6. **Add `@Qualifier("orgRegisterClient")`, configurable HTTP timeout, null-source guard**
   - **`@Qualifier("orgRegisterClient")` missing** — `NetexPublicationDeliveryHttpOrganisationRegistry.java` ctor: ambiguous when sobek-app's authorized `WebClient` is also on the path.
   - **Hard-coded 30s timeout** — same file, `getPublicationDeliverySource()`: make it a property (e.g. `netex.organisations.http-timeout-seconds`).
   - **Null body → null `Source` → uncaught NPE** — same method returns `null`; `loadOrganisations()` then NPEs through `unmarshalFromSource`. Throw a typed exception or return an empty `Source`.
   - Bonus: **`InputStream` leak** — `NetexPublicationDeliveryFileOrganisationRegistry.java` `getPublicationDeliverySource()` wraps a classpath stream in `StreamSource` and never closes it. Read into a `byte[]` and wrap a `ByteArrayInputStream`.

7. **De-duplicate the fixture; split unrelated refactors into a separate PR**
   - **5,109-line fixture duplicated** in both `sobek-app/.../fixtures/organisations-netex-dev.xml` and `sobek-common/.../fixtures/organisations-netex-dev.xml`. Keep one copy in sobek-common and reference from sobek-app via test classpath.
   - **`@Autowired` → `@Inject`** in `sobek-app/.../VehicleImportResource.java` ctor — unrelated, drop.
   - **`PublicationDeliveryPartialUnmarshaller.java`** field/exception cleanup — unrelated, split.
   - **`MultilingualStringMapper` 96+/96-** is a CRLF↔LF flip — restore line endings, keep the rename.
   - **Dead String branch** — `sobek-app/.../OrganisationFetcher.java` `get(...)`: the `else if (orgArg instanceof String s)` branch can't be reached for a typed enum input and `fromValue(unknown)` would throw. Drop it.
   - **Misleading Javadoc** — `NetexPublicationDeliveryOrganisationRegistry.java` `validateOperatorRef` / `validateAuthorityRef`: comments say "all organisations are valid …" but the code correctly filters by `instanceof`. Fix the comments.

8. **Add the missing filter / edge-case tests** — GraphQL test currently only checks the page envelope. Add: filter by `organisationType` (AUTHORITY / OPERATOR / unsupported), filter by `ids` (single, multiple, unknown), combined filter, empty / past-the-end page, and the Authority/Operator id-collision case from #1.

---
*Posted via Claude Code (`/review`).*

File: sobek-repository/src/main/java/org/rutebanken/sobek/repository/OrganisationRepositoryImpl.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+                .map(org -> new Organisation(
+                        org.getId(),
+                        multilingualStringMapper.mapToSobek(org.getName()),
+                        getOrganisationType(org)
+                ))
+                .toList();
+        return organisationList;
+    }
+
+    private OrganisationTypeEnumeration getOrganisationType(Organisation_VersionStructure org) {
```
Comment (dynnamitt) [inline sobek-repository/src/main/java/org/rutebanken/sobek/repository/OrganisationRepositoryImpl.java:80]: jeg tror claude-review no2;
koker ned til denne;
iom Type er en enum med flere verdier
bør den enten også mappe inn disse, eller så bør
navn på metoden sitt navn være
```
a) isOrgAuthorityOrOperatorOrOther() // stupid
b) getOrgCompactedType // ok!
```

Comment (dynnamitt) [reply sobek-repository/src/main/java/org/rutebanken/sobek/repository/OrganisationRepositoryImpl.java:80]: https://entur.github.io/netex-typescript-model/base/netex-schema.html#OrganisationTypeEnumeration

**Review by dynnamitt — APPROVED** [review]
